lib/ContentTypeParser.js

```diff
@@ -28,6 +28,7 @@ ContentTypeParser.prototype.add = function (contentType, opts, parserFn) {
   const parser = new Parser(
     opts.parseAs === 'string',
     opts.parseAs === 'buffer',
+    opts.bodyLimit || null,
```

Comment:
Can we configure here the body limit option and not at line `98`?
I think you can access the global limit also here, or even better in the `addContentTypeParser` api inside `fastify.js`. In this way we can skip that check at runtime, that will likely be falsy very often.

Reply:
Doing so makes it really complicated to also add the default `application/json` parser and not break tests. Basically, it'll require refactoring how the default parser is added and detected.

Reply:
Maybe not. I'm looking into it.

Reply:
Is there a reason we are expecting `OPTIONS` to always reject bodies?

https://github.com/fastify/fastify/blob/034d20f1e2b7c8d5f595a2af1f832558249d5d4a/test/helper.js#L287-L296

The spec allows them:

> If the OPTIONS request includes an entity-body (as indicated by the
   presence of Content-Length or Transfer-Encoding), then the media type
   MUST be indicated by a Content-Type field.

https://tools.ietf.org/html/rfc2616#section-9.2

Reply:
??
That's a `413 - Payload Too Large` test so that test is supposed to reject the body of any request.

Reply:
It's set way up in the test file:

https://github.com/fastify/fastify/blob/034d20f1e2b7c8d5f595a2af1f832558249d5d4a/test/helper.js#L67

Reply:
That's a completely different test. They should not be sharing state.

Reply:
I agree

Reply:
@jsumners tell me if I can help :)